app/grpc/gasestimation/gas_estimator.go

```diff
@@ -149,9 +174,23 @@ func estimateGasPriceForTransactions(gasPrices []float64, priority TxPriority) (
 		}
 		return Median(gasPrices[:bottom10PercentIndex])
 	case TxPriority_TX_PRIORITY_MEDIUM:
-		return Median(gasPrices)
+		estimation, err := Median(gasPrices)
+		if err != nil {
+			return 0, err
+		}
+		if stDev < gasPriceAdjustmentThreshold {
+			return estimation * mediumPriorityGasAdjustmentRate, nil
+		}
+		return estimation, nil
```

Comment:
[nit] this can combine two case statements so that the implementation doesn't have to be duplicated

```go
	case TxPriority_TX_PRIORITY_MEDIUM, TxPriority_TX_PRIORITY_UNSPECIFIED:
```

Reply:
good idea 👍  
9237373